integration/cluster.go

```diff
@@ -711,6 +711,16 @@ func (m *member) StopWithAutoLeaderTransfer(t *testing.T) {
 	plog.Printf("stopped %s (%s)", m.Name, m.grpcAddr)
 }
 
+// checkLeaderTransition waits for leader transition, returning the new leader ID.
+func checkLeaderTransition(t *testing.T, m *member) uint64 {
```

Comment:
this interface is naturally racey, try `func checkLeaderTransition(t *testing.T, m *member, oldLead uint64) uint64`